strider/query_planner.py

```diff
@@ -431,138 +156,33 @@ async def generate_plans(
-        for node_id in current_qg['nodes'].keys():
-            reified_graph[node_id] = []
-        for edge_id in current_qg['edges'].keys():
-            reified_graph[edge_id] = []
-
-        # Fill in adjacencies
-        for edge_id, edge in current_qg['edges'].items():
-            reverse = edge["kp"]["reverse"]
-            if not reverse:
-                # Adjacency for forward edge
-                reified_graph[edge['subject']].append(edge_id)
-                reified_graph[edge_id].append(edge['object'])
-            else:
-                # Adjacency for reverse edge
-                reified_graph[edge['object']].append(edge_id)
-                reified_graph[edge_id].append(edge['subject'])
-
-        # Starting at each pinned node, find possible traversals through
-        # the operation graph
-        possible_traversals = []
-        for pinned in pinned_nodes:
-            possible_traversals.extend(
-                traversals_from_node(reified_graph, pinned)
-            )
-
-        # pylint: disable=cell-var-from-loop
-        possible_traversals = filter(
-            lambda t: ensure_traversal_connected(current_qg, t),
-            possible_traversals)
-
-        possible_traversals = list(possible_traversals)
-
-        for traversal in possible_traversals:
-            # Build our list of steps in the plan
-            plan = defaultdict(list)
-            for edge_id in traversal:
-                # Skip iteration for non-edges
-                if edge_id not in current_qg['edges'].keys():
-                    continue
-
-                # We need to know which way to step through the edge
-                # so we use the previous value in the traversal
-                # which is always the source node
-                edge = current_qg['edges'][edge_id]
-
-                kp = edge["kp"]
-                reverse = kp["reverse"]
-                if reverse:
-                    step = Step(edge['object'], edge_id, edge['subject'])
-                else:
-                    step = Step(edge['subject'], edge_id, edge['object'])
-
-                # Attach information about categorys to kp info
-                plan[step].append(kp)
-            plans.append(plan)
-
-    num_duplicated_plans = len(plans)
-
-    # collapse plans
-    unique_map = defaultdict(lambda: defaultdict(list))
-    for plan in plans:
-        key = tuple(plan.keys())
-        for step, kps in plan.items():
-            for kp in kps:
-                if kp not in unique_map[key][step]:
-                    unique_map[key][step].append(kp)
-    plans = list(unique_map.values())
-
-    logger.info(
-        f"Found {num_duplicated_plans} plans, collapsed down to {len(plans)} plans")
-
-    if len(plans) == 0:
+    kps = dict()
+    for qedge_id in qgraph["edges"]:
+        qedge = qgraph["edges"][qedge_id]
+        provided_by = {"allowlist": None, "denylist": None} | qedge.get("provided_by", {})
+        kp_results = await kp_registry.search(
```

Comment:
You could potentially use the `step_to_kps` function here.

Reply:
023acde943e9a1a53674d5c4ec892482b5109de1